d2l-dom.js

```diff
@@ -72,6 +72,40 @@ D2L.Dom = {
 		return this.findComposedAncestor(node, function(node) {
 			return (node === ancestorNode);
 		}) !== null;
+	},
+
+	getOffsetParent: function(node) {
+		if (!window.ShadowRoot) {
+			return node.offsetParent;
+		}
+
+		if (
+			!this.getComposedParent(node) ||
+			node.tagName === 'BODY' ||
+			window.getComputedStyle(node).position === 'fixed'
```

Comment:
A bunch of things in here are going to [force layout/reflow](https://gist.github.com/paulirish/5d52fb081b3570c81e3a). @dbatiste, do you think we should make it async and use fastdom?

Reply:
Alternatively I guess whoever calls this could know that it's an alternate for `elem.offsetParent` which itself forces layout so they should use fastdom themselves.

Reply:
Yeah think that would be good to leave it to the consumer to use fast-dom since they might want to batch it with other measurements.  I think they could still batch it if we made this async though, but the consumers code might be a little more awkward.

We do use fast-dom a bunch throughout these helpers though - namely the vanilla expand/collapse, arrowkeys-behavior, and visible-on-ancestor - they are all doing higher level stuff though.